src/algebra/big_operators.lean

```diff
@@ -307,8 +307,116 @@ lemma prod_eq_one [comm_monoid β] {f : α → β} {s : finset α} (h : ∀x∈s
 calc s.prod f = s.prod (λx, 1) : finset.prod_congr rfl h
   ... = 1 : finset.prod_const_one
 
+@[to_additive finset.sum_filter]
+lemma prod_filter [decidable_eq α] [comm_monoid β]
+  (s : finset α) (f : α → β) (P : α → Prop) [decidable_pred P] :
+s.prod f = (s.filter P).prod f * (s.filter (λ a, ¬ P a)).prod f :=
+by rw [← finset.prod_union (finset.filter_inter_filter_neg_eq s), finset.filter_union_filter_neg_eq]
+
+-- TODO These next few should probably also be proved without the assumption of commutativity.
+@[to_additive finset.Ico_sum_split]
+lemma Ico_prod_split (l n m : ℕ) (h₁ : n ≤ l) (h₂ : l ≤ m) (f : ℕ → β) :
+(Ico n m).prod f = (Ico n l).prod f * (Ico l m).prod f :=
+begin
+  rw prod_filter (Ico n m) f (λ x, x < l),
+  rw filter_not,
+  simp,
+  rw min_eq_right h₂,
+  rw max_eq_right h₁,
+end
+
+-- `to_additive` chokes on this one, so we reprove it below
+lemma Ico_prod_split_first (n m : ℕ) (h : n < m) (f : ℕ → β) :
+(Ico n m).prod f = f n * (Ico (n+1) m).prod f :=
```

Comment:
```suggestion
  (Ico n m).prod f = f n * (Ico (n+1) m).prod f :=
```

Reply:
Hm, this surely shouldn't use `Ico_prod_split`. Isn't there a specific rule that `Ico` can be represented as an insert on the smaller side?

Reply:
Update coming...